robottelo/hosts.py

```diff
@@ -250,6 +253,32 @@ def power_control(self, state=VmState.RUNNING, ensure=True):
             except TimedOutError:
                 raise ContentHostError('Unable to connect to host that should be running')
 
+    def download_file(self, file_url, local_path=None, file_name=None):
+        """Downloads file from given fileurl to directory specified by local_path by given filename
+        on satellite.
+
+        If remote directory is not specified it downloads file to /tmp/.
+
+        :param str file_url: The complete server file path from where the
+            file will be downloaded.
+        :param str local_path: Name of directory where file will be saved. If not
+            provided file will be saved in /tmp/ directory.
+        :param str file_name: New name of the Downloaded file else its given from file_url
+
+        :returns: Returns list containing complete file path and name of downloaded file.
+        """
+        if not file_name:
+            _, file_name = os.path.split(file_url)
+
+        if not local_path:
+            local_path = '/tmp/'
+
+        # download on server
+        result = self.execute(f'wget -O {local_path}{file_name} {file_url}')
+        if result.status != 0:
+            raise DownloadFileError(f'Unable to download {file_name}')
+        return [f'{local_path}{file_name}', file_name]
```

Comment:
If we use `PurePath`s here we can remove the `os` import and smooth some things out.
```suggestion
        file_name = PurePath(file_name or file_url).name
        local_path = PurePath(local_path or '/tmp') / file_name

        # download on server
        result = self.execute(f'wget -O {local_path.absolute()} {file_url}')
        if result.status != 0:
            raise DownloadFileError(f'Unable to download {file_name}: {result.stderr}')
        return local_path, file_name
```
This would mean the return for this method also changes to be a tuple consisting of the local_path as a PurePath object and the original file_name

Reply:
@JacobCallahan Shouldnt be a blocker comment but since this is failing tests I didnt put much effort in improving code as I said in description of this RP.

Will look into it tomorrow !

Reply:
@JacobCallahan Updated with some fixes from suggestion script and locally lo0ks good ! Also running PRT lets see how it goes !